src/modules/formBuilder/formBuilderUtil.ts

```diff
@@ -214,3 +214,79 @@ export const insertItemToDefinition = ({
     newParentArray.splice(insertAtIndex, 0, item);
   }
 };
+
+/**
+ * Validates that the attempted deletion action is allowed. It is allowed if the item
+ * the user is trying to delete doesn't have any other items that refer to it.
+ * If dependents exist, then the operation is not allowed and the function returns a user-facing error string.
+ *
+ * @param toDelete the linkId of the item the user is trying to delete
+ * @param definition the current form definition
+ */
+export const validateDeletion = ({
```

Comment:
UI suggestion for modal:
```
Items with autofill condition(s) that depend on this item:
* foo
* bar

Items with visibility condition(s) that depends on this item:
* foo

Items with min/max bound(s) that depends on this item:
* foo
```

as opposed to:


<img width="632" alt="Screenshot 2024-06-26 at 2 03 55 PM" src="https://github.com/greenriver/hmis-frontend/assets/5333982/7df14511-f475-4d6a-8b59-9ab31c172dec">


---

Thoughts on that?

That could be accomplished by making this function return the actual dependencies (or their labels) rather than strings to be rendered, so we have some more flexibility in adjusting the alert interface (i.e. adding links to those items later). It also could be helpful if we ever want to add links to dependent items into the property editor drawer.

```
const { autofillDependents, enableWhenDependents, boundDependents } = getDependentItems(itemMap, linkId)
```

Reply:
What about:
```
Cannot delete item

[Item Title] cannot be deleted because it is referenced elsewhere:
* Items with autofill condition(s):
  * [Other Item Title]
  * [Other Item Title]
* Items with visibility condition(s):
  * [Other Item Title]
* Items with min/max bound(s):
  * [Other Ix`tem Title]

All references to this item must be removed before it can be deleted
```

The first and last sentence may be redundant, but since it's instructions on how to resolve the issue it might benefit from some repetition. Also this doesn't account for questions with the same titles.

Reply:
Great, I think these are both big improvements.

I think often there will be only one dependent item -- should the display change when that's the case? such as removing all the nesting and just saying,

> [Item Title] cannot be deleted because [Other Item Title]'s autofill condition(s) depend on it.

Agree about bold vs. quotes. Which do you prefer? I uploaded 2 quick screenshots to [figjam](https://www.figma.com/board/axGI9vwmuBRC4uZ11ZJ3RP/Form-Builder-Workflow?node-id=900-7646&t=LPqSP0K6FaR93jiY-0) so we can compare. (Sorry for the lack of bullet points, I still need to add them)

I'm not 100% sure about "it is referenced" -- the grammarian in me (lol) wants to say that the verb is "to refer" rather than "to reference" but I guess "to reference" is also common usage nowadays. Maybe "because it is referred to elsewhere" or "because other item(s) refer to it"?